### ISSUE TITLE
Edit screen summary should be grouped in a list

### ISSUE THREAD

**Issue by amberhinds** [equalizedigital/accessibility-checker#359]
<img width="944" alt="Screenshot 2023-11-20 at 9 40 03 AM" src="https://github.com/equalizedigital/accessibility-checker/assets/13153456/20d49fda-dbf6-45af-9578-5930994ae7b0">

This should be an unordered list. The numbers can be styled larger with spans, but each tile should be a list item. 

Also add an area-label on the li that has the count and label - I.e., aria-label="19 warnings" - because the different rows are confusing on a screen reader.
